lib/models/BackendInfo.ts

```diff
@@ -52,7 +52,8 @@ export default class BackendInfo {
         locationConstraint: string | undefined,
         log: RequestLogger,
     ) {
-        if (!locationConstraint || !(locationConstraint in config.locationConstraints)) {
+        if (!locationConstraint || !(locationConstraint in config.locationConstraints) ||
```

Comment:
maybe easier to read this way?

```suggestion
        if (!config.locationConstraints[locationConstraint] ||
```

Reply:
Can't really do that in typescript as `locationConstraint` can be undefined.